src/components/file/file.service.ts

```diff
@@ -209,17 +209,30 @@ export class FileService {
     { parentId, uploadId, name }: CreateFileVersionInput,
     session: ISession
   ): Promise<File> {
-    let upload;
-    try {
-      upload = await this.bucket.headObject(`temp/${uploadId}`);
-    } catch (e) {
-      if (
-        (e as AWSError).code === 'NotFound' ||
-        e instanceof NotFoundException
-      ) {
-        throw new InputException('Could not find upload', 'uploadId');
+    const [tempUpload, existingUpload] = await Promise.allSettled([
+      this.bucket.headObject(`temp/${uploadId}`),
+      this.bucket.headObject(uploadId),
+    ]);
+
+    if (
+      tempUpload.status === 'rejected' &&
```

Comment:
I think we should also verify the type of the error in all of these cases. The previous code only threw that exception if it determined the error/reason was not found. Otherwise it threw a `ServerException`.